### DISCUSSION TITLE
Add additional index to assettransformindexes

### DISCUSSION THREAD

**Discussion by lenvanessen** [craftcms/cms#10427] [Ideas] [answered]
Hi There,

Another performance improvement we can implement! 

If you eager to load the transforms, like suggested here: https://craftcms.com/docs/3.x/dev/eager-loading-elements.html#eager-loading-image-transform-indexes It will trigger this query:

```
SELECT `id`, `assetId`, `filename`, `format`, `location`, `volumeId`, `fileExists`, `inProgress`, `error`, `dateIndexed`, `dateUpdated`, `dateCreated`
FROM `assettransformindex`
WHERE (`assetId` IN (322461, 320454)) AND (((`location`='_author') AND (`format`='jpg')) OR ((`location`='_authorLarge')
``` 

The `assettransformindex` doesn't have a combined key for `assetId,format,location`. So the query is around 80ms for our system. After manually adding this index, the query only took 8ms.

Since it's standard Craft behavior, I believe this would also be a nice performance improvement

**Comment by brandonkelly** [comment]
To be clear, are the columns in that specific order in your custom index? (`assetId,format,location`)

**Reply by lenvanessen** [reply]
@brandonkelly  yes!

**Reply by brandonkelly** [reply]
Thanks! Added for the next release.

**Comment by brandonkelly** [comment, marked as answer]
Craft 3.7.32 is out now with this change.

**Reply by moondawg69** [reply]
Unfortunately, just updated to this, it thrown a 500 error on new transforms. Had to roll back to the prior Craft version. 

Error 500 - tried all the usual cache clears to asset index rebuild - no other error messages in storage etc.

Erm help :)

**Reply by lenvanessen** [reply]
@moondawg69 if the web.log is empty, what does your server log say? And by "new transforms", do you mean adding a new transform from the settings in craft?

**Reply by moondawg69** [reply]
@lenvanessen - sorry, looks like an environment error on the server, for some reason, and reached out to my managed support:

"The `webp` format is not supported on this server! "

Is being returned. Oh joy ;)

Sorry for wasting your time.

**Reply by lenvanessen** [reply]
@moondawg69 ha, webp support is bad unfortunately with many hosting companies unfortunately..

**Reply by iamkeir** [reply]
@moondawg69 webp was working fine for us until recently, now experiencing [same "webp format is not supported" issue](https://github.com/craftcms/cms/issues/10588). Did you find the cause/a fix?

**Reply by moondawg69** [reply]
@iamkeir yes, switched off imagick and moved to GD. Instant fix. Summary of the environment now:

PHP version | 8.0.15
Linux 3.10.0-962.3.2.lve1.5.60.el7.x86_64
MySQL 5.5.5
GD 8.0.15

Hope that helps :)

**Reply by iamkeir** [reply]
@moondawg69 yep, thanks, that fixed it for us too, wonder why that issue has just popped up with Imagick.

**Reply by moondawg69** [reply]
@iamkeir I had a quick look around, something funky with Cloud Linux/Cent OS and imagick/webp using v old libraries...
